stellargraph/mapper/corrupted.py

```diff
@@ -171,7 +171,7 @@ def corrupt_group(group_idx, group):
         ]
 
         # create the appropriate labels
-        batch_size = inputs[0].shape[: self.num_batch_dims]
+        batch_size = min(inp.shape[: self.num_batch_dims] for inp in inputs)
```

Comment:
Hm, so this works because it's picking up the output indices tensor with shape `(1, num_outputs)`, right?

Maybe we could expand the comment to discuss this briefly? E.g.

```python
# create the appropriate labels: this needs to match the number of outputs in the batch. 
# For full batch methods, this is the output indices tensor (batch shape = 1 × num_out), 
# not the features tensor (1 × num_nodes). Rather than hard-code, we assume that the
# smallest "batch shape" is the right one.
```

We could potentially also rename this like the following, to hammer home the point:

```suggestion
        output_batch_shape = min(inp.shape[: self.num_batch_dims] for inp in inputs)
```

Reply:
> Hm, so this works because it's picking up the output indices tensor with shape (1, num_outputs)

Yep that's right, and for SAGE algos all inputs have the same batch shape anyways so we're all good (for now...)

> Maybe we could expand the comment to discuss this briefly?

Yeah I think that's a could point, its a strange and subtle bit of code that will probably trip someone up in the future.